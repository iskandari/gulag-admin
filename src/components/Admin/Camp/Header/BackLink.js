import styled from 'styled-components';
import { Link } from 'react-router-dom';

export default styled(Link)`
  font-size: 12px;
  font-weight: bold;
  color: #000;
  text-decoration: none;
  transition: .2s;
  &:hover {
    opacity: .5;
    transition: .2s;
  }
`;
