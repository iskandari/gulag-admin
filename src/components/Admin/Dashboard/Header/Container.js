import styled from 'styled-components';

export default styled.header`
  display: flex;
  justify-content: space-between;
  align-items: flex-start;
  
  padding-top: 40px;
  margin-bottom: 50px;
`;
