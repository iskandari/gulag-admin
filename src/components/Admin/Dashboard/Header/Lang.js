import styled from 'styled-components';

export default styled.div`
  display: inline-block;
  margin-right: 15px;

  font-weight: 900;

  &:last-child {
    margin-right: 0;
  }
`;
