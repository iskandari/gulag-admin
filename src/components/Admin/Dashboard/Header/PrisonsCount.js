import styled from 'styled-components';

export default styled.div`
  font-size: 36px;

  & span {
    display: inline-block;
    padding-top: 3px;

    vertical-align: top;

    font-size: 14px;

    opacity: 0.5;
  }
`;
