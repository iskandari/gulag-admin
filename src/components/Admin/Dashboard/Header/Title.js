import styled from 'styled-components';

export default styled.div`
  padding: 6px 0;
  border-bottom: 1px solid rgba(0, 0, 0, .3);
  margin-bottom: 10px;
`;
