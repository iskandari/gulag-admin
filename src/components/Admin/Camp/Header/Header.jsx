import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

// styled
import Container from './Container';
import BackLink from './BackLink';
import Title from './Title';
import Subtitle from './Subtitle';
import Button from '../../Button';
import PrisonSaved from './PrisonSaved';

// utils
import { getPeriods } from '../../../../utils/utils';

const Header = ({ camp, updateCamp }) => (
  <Container>
    <BackLink to='/admin'>← к таблице лагерей</BackLink>
    <div>
      <Title>{camp.getIn(['name', 'ru'])}</Title>
      <Subtitle>{getPeriods(camp.toJS())}</Subtitle>
    </div>
    <div>
      <Button
        color='orange'
        onClick={updateCamp}
      >
        сохранить
      </Button>
      <PrisonSaved>
        <div>Сохранено:</div>
        <span>
          {moment(camp.get('updated_at'))
            .locale('ru')
            .format('DD.MM.YY, HH:mm')}
        </span>
      </PrisonSaved>
    </div>
  </Container>
);

Header.propTypes = {
  camp: PropTypes.object.isRequired,
  updateCamp: PropTypes.func.isRequired
};

export default Header;
