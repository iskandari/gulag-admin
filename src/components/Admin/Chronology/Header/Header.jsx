import React from 'react';
import PropTypes from 'prop-types';

// styled
import Container from './Container';
import BackButton from './BackButton';
import Title from './Title';
import SaveButton from './SaveButton';

const Header = ({ goBack, updatePeriods }) => (
  <Container>
    <BackButton onClick={goBack}>
      ← к таблице лагерей
    </BackButton>
    <Title>
      Хронология ГУЛАГа
    </Title>
    <div>
      <SaveButton onClick={updatePeriods}>
        сохранить
      </SaveButton>
    </div>
  </Container>
);

Header.propTypes = {
  goBack: PropTypes.func.isRequired,
  updatePeriods: PropTypes.func.isRequired
};

export default Header;
