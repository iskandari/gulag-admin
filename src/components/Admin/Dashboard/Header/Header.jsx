import React from 'react';
import PropTypes from 'prop-types';

// styled
import Container from './Container';
import Button from '../../Button';
import Title from './Title';
import Lang from './Lang';
import PrisonsCount from './PrisonsCount';

const Header = props => {
  const {
    campsCount,
    publishedRuCount,
    publishedEnCount,
    publishedDeCount,
    logout,
    createCamp
  } = props;

  return (
    <Container>
      <Button onClick={logout}>
        Выйти
      </Button>
      <div>
        <Title>
          Опубликовано лагерей
        </Title>
        <div>
          <Lang>
            <PrisonsCount>
              {publishedRuCount}
              <span>
                {`/${campsCount}`}
              </span>
            </PrisonsCount>
            На русском
          </Lang>
          <Lang>
            <PrisonsCount>
              {publishedEnCount}
              <span>
                {`/${campsCount}`}
              </span>
            </PrisonsCount>
            На английском
          </Lang>
          <Lang>
            <PrisonsCount>
              {publishedDeCount}
              <span>
                {`/${campsCount}`}
              </span>
            </PrisonsCount>
            На немецком
          </Lang>
        </div>
      </div>
      <Button color='orange' onClick={createCamp}>
        Добавить лагерь
      </Button>
    </Container>
  );
};

Header.propTypes = {
  campsCount: PropTypes.number.isRequired,
  publishedRuCount: PropTypes.number.isRequired,
  publishedEnCount: PropTypes.number.isRequired,
  publishedDeCount: PropTypes.number.isRequired,
  logout: PropTypes.func.isRequired,
  createCamp: PropTypes.func.isRequired
};

export default Header;
