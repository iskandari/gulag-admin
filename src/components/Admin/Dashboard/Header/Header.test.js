import React from 'react';
import renderer from 'react-test-renderer';

import Header from './Header';

it('renders', () => {
  const renderedValue = renderer.create(<Header />)
    .toJSON();
  expect(renderedValue)
    .toMatchSnapshot();
});
