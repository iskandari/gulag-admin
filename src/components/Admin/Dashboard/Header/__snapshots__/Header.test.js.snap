// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<header
  className="sc-bdVaJa iEbmKW"
>
  <button
    className="sc-bwzfXH cjRVLf"
    onClick={undefined}
  >
    Выйти
  </button>
  <div>
    <div
      className="sc-htpNat oVgFz"
    >
      Опубликовано лагерей
    </div>
    <div>
      <div
        className="sc-bxivhb gsTOhO"
      >
        <div
          className="sc-ifAKCX RqEBc"
        >
          <span>
            /
          </span>
        </div>
        На русском
      </div>
      <div
        className="sc-bxivhb gsTOhO"
      >
        <div
          className="sc-ifAKCX RqEBc"
        >
          <span>
            /
          </span>
        </div>
        На английском
      </div>
      <div
        className="sc-bxivhb gsTOhO"
      >
        <div
          className="sc-ifAKCX RqEBc"
        >
          <span>
            /
          </span>
        </div>
        На немецком
      </div>
    </div>
  </div>
  <button
    className="sc-bwzfXH cZtovU"
    color="orange"
    onClick={undefined}
  >
    Добавить лагерь
  </button>
</header>
`;
